Sources/Build/BuildManifest/LLBuildManifestBuilder+Swift.swift

```diff
@@ -191,10 +191,8 @@ extension LLBuildManifestBuilder {
     public func addTargetsToExplicitBuildManifest() throws {
         // Sort the product targets in topological order in order to collect and "bubble up"
         // their respective dependency graphs to the depending targets.
-        let nodes = self.plan.targets.compactMap {
-            ResolvedModule.Dependency.module($0.module, conditions: [])
-        }
-        let allPackageDependencies = try topologicalSort(nodes, successors: { $0.dependencies })
+        let allPackageDependencies = self.plan.targets.flatMap { $0.recursiveDependencies(using: self.plan) }
```

Comment:
@MaxDesiatov This is something I want to draw your attention to - previously `allPackageDependencies` didn't check conditions associated with dependencies but `recursiveDependencies(using:)` does that and I think that's reasonable for what this logic is trying to archive.

Reply:
Seems reasonable on the surface. Does this change some observable behavior that we should add a test for?

Reply:
It doesn't that's why I wanted to bring it up.

Reply:
Even if I show up in `git blame` for that line, I'm pretty sure I didn't write the original logic or change it at any recent point, it was probably some NFC refactor or reformatting. I don't mind this specific change, it makes sense, there's no significant readability regression, and it seems to be NFC

Reply:
Yeah, no worries!